src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,54 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 1;
+    private final int[] array;
 
     public MergeSortAction(int[] array) {
+        this.array = array;
+    }
+
+    @Override
+    protected void compute() {
+        int start = 0;
+        int workLoad = array.length;
+
+        if (workLoad > THRESHOLD) {
+            int middle = workLoad / 2;
```

Comment:
The 'middle' variable is calculated based on 'workLoad', which is the length of the array. Since 'start' is always 0, this is fine, but if 'start' is meant to be used in the future, you would need to adjust the calculation of 'middle' to account for the starting position.